File: base/src/main/java/io/quarkus/code/rest/DownloadRerouteFilter.java
package io.quarkus.code.rest;

import io.vertx.core.http.HttpMethod;
import io.vertx.ext.web.Router;
import jakarta.enterprise.context.ApplicationScoped;
import jakarta.enterprise.event.Observes;

@ApplicationScoped
public class DownloadRerouteFilter {

    public void init(@Observes Router router) {
        router.route(HttpMethod.GET, "/d").handler(ctx -> {
            ctx.reroute(ctx.request().uri().replace("/d", "/api/download"));
        });
    }
}

File: api/src/main/kotlin/io/quarkus/code/rest/DownloadRerouteFilter.kt
package io.quarkus.code.rest

import io.vertx.core.http.HttpMethod
import io.vertx.ext.web.Router
import javax.enterprise.context.ApplicationScoped
import javax.enterprise.event.Observes

@ApplicationScoped
class DownloadRerouteFilter {

    fun init(@Observes router: Router) {
        router.route(HttpMethod.GET,"/d").handler {
            it.reroute(it.request().uri().replace("/d", "/api/download"))
        }
    }

}